BinaryTreeDepthRecursive.java

```diff
@@ -0,0 +1,45 @@
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class BinaryTreeDepthRecursive {
+    public static void main(String[] args) {
+        test(new int[]{1, 2, 3, 4, 5, 6, 7, 8});
+        test(new int[]{});
+        test(new int[]{1});
+        test(new int[]{1, 2});
+    }
+
+    public static int findDepth(Node root) {
+        if(root == null) return 0;
+        if (root.left == null && root.right == null) return 0;
```

Comment:
The more correct way of doing this is to return a depth of -1 when root == null, and don't even bother definining the second base case.